process/block/sovereignChainBlock.go

```diff
@@ -1557,9 +1557,13 @@ func (scbp *sovereignChainBlockProcessor) updateState(header data.HeaderHandler,
 	scbp.blockChain.SetFinalBlockInfo(header.GetNonce(), headerHash, header.GetRootHash())
 }
 
-// TODO: (sovereign) remove these cleanUpFunc pointer functions once shardCoordinator will return the correct shard id from task: MX-14132
-func (scbp *sovereignChainBlockProcessor) cleanupBlockTrackerPoolsForShard(_ uint32, noncesToPrevFinal uint64) {
-	scbp.baseCleanupBlockTrackerPoolsForShard(core.MainChainShardId, noncesToPrevFinal)
+func (scbp *sovereignChainBlockProcessor) cleanupBlockTrackerPoolsForShard(shardID uint32, noncesToPrevFinal uint64) {
```

Comment:
the previous todo said:
`// TODO: (sovereign) remove these cleanUpFunc pointer functions once shardCoordinator will return the correct shard id from task: MX-14132`
why is not the correct shardID sent here? 
the sovereign should not have anything to do with the metachain, right?

Reply:
The shardID comes from the base implementation which uses hardcoded MetachainShardId which is not ideal. 
I added back the TODO for the shardCoordinator.

Reply:
Because this func is called from another base processing path to cleanup cross meta headers.
Part of that `TODO` is fixed for selfShardID not(which is 0 == SovereignChainHeader), but the other one needs extra work to fix, perhaps in another PR